### PR TITLE
Added `Switch` Animation for Material 3

### LINKED ISSUE #103536: Update `Switch` to support Material 3
As part of the [Material 3 specification](https://m3.material.io/components/switch/overview) there are updates to the Switch's visuals:

Screen Shot 2022-05-11 at 1 53 27 PM

We need to update the existing `Switch` class in the Flutter Material library to support this.

### LINKED ISSUE #111663: Material 3 Switch thumb size and motion isn't quite to spec
https://github.com/flutter/flutter/issues/103536#issuecomment-1243481591

Expected | Actual |
--- | --- | 
image | image

### LINKED ISSUE #112026: Switch thumb radius decreases by 1 px when it moves to the middle.
The thumb radius in Material 2 Switch decreases a little when it moves to the middle of the track. But in Material 3, the thumb radius should stay the same.

Expected:
https://user-images.githubusercontent.com/36861262/191396770-b8fc0267-dfdf-4460-b885-051b085346cf.MP4
Screen Shot 2022-09-20 at 7 06 50 PM

Actual:
https://user-images.githubusercontent.com/36861262/191396754-6523ac57-d9e5-4b85-a858-9cc77f2f21c3.MP4
Screen Shot 2022-09-20 at 7 06 14 PM

### PR DESCRIPTION
**Pull request by QuncCccccc**
Part of: https://github.com/flutter/flutter/issues/91605

This PR is to add animation for M3 `Switch`. 

https://user-images.githubusercontent.com/36861262/194502373-2d8c290f-13b1-4258-a373-ba377dc38e6c.mov

Slowed down for reference:

https://user-images.githubusercontent.com/36861262/194502598-7501d83c-d6b9-45be-a589-46d1ff546fa6.mov

Fixes #103536
Fixes #112026
Fixes #111663

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by QuncCccccc** [comment]
I just marked the golden change as positive since it is a new test for the Switch thumb in the middle of the track.

**Review by HansMuller — APPROVED** [review]
Nice work! LGTM

File: dev/tools/gen_defaults/lib/switch_template.dart [hunk 1]
```diff
@@ -202,6 +202,19 @@ class _SwitchConfigM3 with _SwitchConfig {
 
   @override
   double get trackWidth => ${tokens['md.comp.switch.track.width']};
+
+  // The thumb size at the middle of the track. Hand coded default based on the animation specs.
+  // https://direct.googleplex.com/#/spec/283220003&285130003
```
Comment (darrenaustin) [inline dev/tools/gen_defaults/lib/switch_template.dart:207]: Is this a publicly available link? If not we probably shouldn't include it.

**Review by darrenaustin — APPROVED** [review]
LGTM. Nice work!

File: dev/tools/gen_defaults/lib/switch_template.dart (hunk 1 above)
Comment (guidezpl) [reply dev/tools/gen_defaults/lib/switch_template.dart:207]: +1

**Comment by guidezpl** [comment]
Great work!

File: dev/tools/gen_defaults/lib/switch_template.dart (hunk 1 above)
Comment (QuncCccccc) [reply dev/tools/gen_defaults/lib/switch_template.dart]: I see. Removed it. Thanks!

**Comment by MaxYablochkin** [comment]
@QuncCccccc , beautiful animation!
For an enabled switch, when the ball hits the border of the switch, the background color of the switch is repainted to a darker shade, and when the ball returns from the border to its original enabled position, the background is repainted to a lighter shade, and this happens on the last frames of the animation. The same applies to the animation of turning off the switch, as well as the shade of the ball itself.
Is this a unique feature, or is it an unwanted result?

**Pull request merged**